app/src/main/java/com/infomaniak/swisstransfer/ui/screen/onboarding/components/AnimatedOnboardingButton.kt

```diff
@@ -0,0 +1,110 @@
+ */
+package com.infomaniak.swisstransfer.ui.screen.onboarding.components
+
+import androidx.compose.animation.core.animateDpAsState
+import androidx.compose.animation.core.animateFloatAsState
+import androidx.compose.animation.core.tween
+import androidx.compose.foundation.layout.*
+import androidx.compose.material3.Icon
+import androidx.compose.material3.Text
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.alpha
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import com.infomaniak.swisstransfer.R
+import com.infomaniak.swisstransfer.ui.components.SwissTransferButton
+import com.infomaniak.swisstransfer.ui.images.AppImages
+import com.infomaniak.swisstransfer.ui.images.icons.ArrowRight
+import com.infomaniak.swisstransfer.ui.theme.Dimens
+import com.infomaniak.swisstransfer.ui.theme.Margin
+import com.infomaniak.swisstransfer.ui.theme.SwissTransferTheme
+import com.infomaniak.swisstransfer.ui.utils.PreviewLightAndDark
+
+private const val BUTTON_ANIM_DURATION = 300
+private const val BUTTON_ANIM_DELAY = BUTTON_ANIM_DURATION / 2
+private const val BUTTON_ANIM_NO_DELAY = 0
+private val FAB_SIZE = 64.dp
+
+@Composable
+fun AnimatedOnboardingButton(isExpanded: () -> Boolean, onClick: () -> Unit) {
+    val buttonWidth by animateDpAsState(
+        targetValue = if (isExpanded()) Dimens.SingleButtonMaxWidth else FAB_SIZE,
+        animationSpec = tween(durationMillis = BUTTON_ANIM_DELAY),
+        label = "Onboarding button width",
+    )
+
+    val buttonHeight by animateDpAsState(
+        targetValue = if (isExpanded()) Dimens.LargeButtonHeight else FAB_SIZE,
+        animationSpec = tween(durationMillis = BUTTON_ANIM_DELAY),
+        label = "Onboarding button height",
+    )
+
+    val arrowDelay = if (isExpanded()) BUTTON_ANIM_NO_DELAY else BUTTON_ANIM_DELAY
+    val arrowVisibility by animateFloatAsState(
+        targetValue = if (isExpanded()) 0f else 1f,
+        animationSpec = tween(durationMillis = BUTTON_ANIM_DELAY, delayMillis = arrowDelay),
+        label = "Onboarding arrow visibility",
+    )
+
+    val textDelay = if (isExpanded()) BUTTON_ANIM_DELAY else BUTTON_ANIM_NO_DELAY
+    val textVisibility by animateFloatAsState(
+        targetValue = if (isExpanded()) 1f else 0f,
+        animationSpec = tween(durationMillis = BUTTON_ANIM_DELAY, delayMillis = textDelay),
+        label = "Onboarding text visibility",
+    )
+
+    SwissTransferButton(
+        modifier = Modifier
+            .height(buttonHeight)
+            .padding(horizontal = Margin.Medium)
+            .width(buttonWidth),
+        onClick = onClick,
+        contentPadding = PaddingValues(),
+    ) {
+        Box(contentAlignment = Alignment.Center) {
+            Icon(
+                modifier = Modifier.alpha(arrowVisibility),
+                imageVector = AppImages.AppIcons.ArrowRight,
+                contentDescription = null
```

Comment:
Doesn't this should be something like "next" ? 
Also it misses a comma

Reply:
Very true indeed, I'll add a content description